static/js/filter.js

```diff
@@ -1,430 +1,176 @@
+            // "active-aliases" – Only show the aliases that are enabled
+            // "disabled-aliases"– Only show the aliases that are disabled
+            // "relay-aliases"– Only show aliases that have been generated from the dashboard/add-on 
+            // "domain-aliases"– Only show aliases that were created with a unique subdomain. 
+            const matchesCategoryFilters = (
+                activeCategoryFilters.length === 0 ||
+                (activeCategoryFilters.includes("active-aliases") && aliasContainer.classList.contains("is-enabled")) ||
+                (activeCategoryFilters.includes("disabled-aliases") && !aliasContainer.classList.contains("is-enabled")) ||
+                (activeCategoryFilters.includes("relay-aliases") && aliasContainer.classList.contains("is-relay-alias")) ||
+                (activeCategoryFilters.includes("domain-aliases") && aliasContainer.classList.contains("is-domain-alias"))
+            );
+
+            if (matchesSearchFilter && matchesCategoryFilters) {
+                aliasContainer.classList.remove("is-hidden");
+            } else {
+                aliasContainer.classList.add("is-hidden");
             }
-        }
-
-        // Show aliases with labels that match the search query
-        for (const result of matchListAliasLabels) {        
-            result.alias.classList.remove("is-hidden");
-        }
-
-	}
-
-    // This function catches any label updates from the user and updates the search array to display them.
-    function updateAliasLabel(event){
-        const alias = event.target.closest(".js-alias");
-        const prevLabel = event.target.dataset.label;
-        const prevLabelLowercased = prevLabel.toString().toLowerCase();
-        const newLabel = event.target.value;
-        const newLabelLowercased = newLabel.toString().toLowerCase();
-        const AliasWithLabelsArrayIndex = aliasesWithLabelsCollection.indexOf(alias);
-        const labelArrayIndex = filterAliasLabels.indexOf(prevLabelLowercased);
+        });
 
+        filterLabelVisibleCases.textContent = aliasContainers.filter(aliasContainer => !aliasContainer.classList.contains("is-hidden")).length;
+        filterLabelTotalCases.textContent = aliasContainers.length;
 
-        // Case: User did not enter any label, nor was one previously set
-        if (prevLabel === "" && newLabel === "") {
-            return;
-        }
-
-        // Case: User deleted/cleared an existing label
-        if (newLabel === "") {
-            filterAliasLabels.splice(labelArrayIndex, 1);
-            aliasesWithLabelsCollection.splice(AliasWithLabelsArrayIndex, 1);
-            return;
+        if (activeSearchFilter.length > 0) {
+            filterSearchInput.classList.add("is-filtered");
+            filterContainer.classList.add("is-filtered-by-search");
+        } else {
+            filterSearchInput.classList.remove("is-filtered");
+            filterContainer.classList.remove("is-filtered-by-search");
```

Comment:
nit: add context comment on this use-case for this if statement 
```suggestion
        // Run if user has input a query into the search form
        if (activeSearchFilter.length > 0) {
            filterSearchInput.classList.add("is-filtered");
            filterContainer.classList.add("is-filtered-by-search");
        } else {
            filterSearchInput.classList.remove("is-filtered");
            filterContainer.classList.remove("is-filtered-by-search");
```

Reply:
Fixed in cfe92c0b7f7a272fb2c10feb32175c75cb9f5cd0.